### PR TITLE
DOC: Add section on intake-esm keys

### PR DESCRIPTION
**Pull request by mgrover1**
## Change Summary

This adds a new how to section on dealing with intake-esm keys, how to get more information about them, and how to change them!

## Related issue number

- Towards #407 

## Checklist

- [x] Tests pass on CI
- [x] Documentation reflects the changes where applicable

### CODE REVIEW

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+---
+jupytext:
+  text_representation:
+    format_name: myst
+kernelspec:
+  display_name: Python 3
+  name: python3
+---
+
+# Understanding Intake-ESM Keys and How to Use Them
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:10]: I'm in the process of only capitalizing the first word in the title and leaving the rest in lower case... 
```suggestion
# Understanding intake-ESM keys and how to use them
```

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+kernelspec:
+  display_name: Python 3
+  name: python3
+---
+
+# Understanding intake-ESM keys and how to use them
+
+Intake-ESM helps with aggregating your datasets using some `keys`. Here, we dig into what exactly these keys are, how they are constructed, and how you can change them. Understanding how this work will help you control how your datasets are merged together, and remove the mystery behind these strings of text.
+
+## Imports and Spin Up a Dask Cluster
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:14]:
```suggestion
## Import packages and spin up a Dask cluster
```

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+We start first with importing `intake` and a `LocalCluster` from `dask.distributed`
+
+```{code-cell} ipython3
+import intake
+from distributed import LocalCluster
+
+cluster = LocalCluster()
+```
+
+## Investigate a CMIP6 Catalog
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:25]:
```suggestion
## Investigate a CMIP6 catalog
```

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+Typically, the next step would be to search and load your datasets using `.to_dataset_dict()` or `.to_datatree()`
+
+```{code-cell} ipython3
+catalog_subset = catalog.search(variable_id='ua')
+dsets = catalog_subset.to_dataset_dict()
+print(dsets)
+```
+
+### Investigating the Keys
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:43]:
```suggestion
### Investigating the keys
```

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+activity_id --> institution_id --> source_id --> experiment_id --> table_id --> grid_label
+```
+
+A more clear of taking a look at these aggregation variables using the `.keys_info()` method for the catalog:
+
+```{code-cell} ipython3
+catalog_subset.keys_info()
+```
+
+## Change our Groupby/Aggregation Controls
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:77]:
```suggestion
## Change our groupby/aggregation controls
```

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+```
+activity_id --> institution_id --> source_id --> experiment_id --> table_id --> grid_label --> member_id
+```
+
+```{code-cell} ipython3
+catalog_subset.keys_info()
+```
+
+### Load our Datasets with the New Keys
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:103]:
```suggestion
### Load our datasets with the new keys
```

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+```
+
+And the .to_dask() method to load our dataset into our notebook.
+
+```{code-cell} ipython3
+ds = catalog_subset[first_key].to_dask()
+ds
+```
+
+### Compare this Dataset with the Original Catalog Configuration
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:125]:
```suggestion
### Compare this dataset with the original catalog configuration
```

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+
+Let's start with a sample CMIP6 catalog! This is a fairly large dataset.
+
+```{code-cell} ipython3
+url ="https://raw.githubusercontent.com/intake/intake-esm/main/tutorial-catalogs/GOOGLE-CMIP6.json"
+catalog = intake.open_esm_datastore(url)
+catalog.df.head()
+```
+
+Typically, the next step would be to search and load your datasets using `.to_dataset_dict()` or `.to_datatree()`
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:35]:
```suggestion
Typically, the next step would be to search and load your datasets using {py:meth}`~intake_esm.core.esm_datastore.to_dataset_dict` or {py:meth}`~intake_esm.core.esm_datastore.to_datatree`
```

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+
+Let's go back to our data catalog... and find these fields. You'll notice they are all column labels! These are key components of the metadata.
+
+```{code-cell} ipython3
+catalog_subset.df
+```
+
+## Using keys_info()
+
+These groupby attributes are columns in our catalog! This means that the datasets which will be merged using the hierarchy:
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:65]:
```suggestion
These groupby attributes are columns in our catalog! This means that the datasets which will be aggregated using the hierarchy:
```

File: docs/source/how-to/understand-keys-and-how-to-change-them.md (hunk 8 above)
Comment (mgrover1) [reply docs/source/how-to/understand-keys-and-how-to-change-them.md:35]: Good catch here!

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+```{code-cell} ipython3
+ds = catalog_subset[first_key].to_dask()
+ds
+```
+
+### Compare this dataset with the original catalog configuration
+
+Compare this to our original catalog, which aggregated one level higher, placing all of the `member_id`s into the same dataset.
+
+**Notice how our metadata now mentions there are 65 member_ids in this dataset, compared to 1 in the previous dataset**
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:129]:
````suggestion
```{note}
Notice how our metadata now mentions there are 65 member_ids in this dataset, compared to 1 in the previous dataset
```
````

File: docs/source/how-to/understand-keys-and-how-to-change-them.md [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+
+## Import packages and spin up a Dask cluster
+
+We start first with importing `intake` and a `LocalCluster` from `dask.distributed`
+
+```{code-cell} ipython3
+import intake
+from distributed import LocalCluster
+
+cluster = LocalCluster()
```
Comment (andersy005) [inline docs/source/how-to/understand-keys-and-how-to-change-them.md:22]:
```suggestion
from distributed import Client

cluster = Client()
```

Comment (mgrover1) [reply docs/source/how-to/understand-keys-and-how-to-change-them.md:22]: Yeah - good point. This is a better approach

**Review by andersy005 — APPROVED** [review]
This looks solid, @mgrover1! Thank you

**Pull request merged**